capa/features/extractors/viv/insn.py

```diff
@@ -81,6 +83,15 @@ def extract_insn_api_features(fh: FunctionHandle, bb, ih: InsnHandle) -> Iterato
         if f.vw.getFunctionMeta(f.va, "Thunk"):
             return
 
+    # Added a case for catching basic blocks that contain direct calls to system functions.
+    if insn.mnem in ("int", "syscall"):
+        if insn.mnem != "int" or insn.opers[0].imm == 128:
+            name = get_library_function_name(f.vw, bb)
```

Comment:
pass the instruction here instead of `bb`

Reply:
is this called per instruction, but then enumerates the full basic block again and again?